src/kubernetes_cluster/proof/controller_runtime_safety.rs

```diff
@@ -99,6 +99,20 @@ pub proof fn lemma_always_resp_matches_at_most_one_pending_req<T>(reconciler: Re
     init_invariant::<State<T>>(sm_spec(reconciler), init(reconciler), stronger_next, invariant);
 }
 
+pub proof fn lemma_always_forall_resp_matches_at_most_one_pending_req<T>(reconciler: Reconciler<T>, cr_key: ObjectRef)
+    requires
+        cr_key.kind.is_CustomResourceKind(),
+    ensures
+        sm_spec(reconciler).entails(tla_forall(|msg| always(lift_state(resp_matches_at_most_one_pending_req(msg, cr_key))))),
```

Comment:
Would `spec_entails_tla_forall` simplify this proof?

Reply:
Yes you will need assert forall msg, but I guess you don't have to do that for ex. Something like:
```
let m_to_p = |msg| always(lift_state(resp_matches_at_most_one_pending_req(msg, cr_key)));
assert forall |msg| sm_spec(reconciler).entails(m_to_p(msg)) by {
  lemma_always_resp_matches_at_most_one_pending_req::<T>(reconciler, msg, cr_key);
};
spec_entails_tla_forall(sm_spec(reconcile), m_to_p);
```

Reply:
Yes, this works! But I can't see this simplifies the proof. We don't have to make `entails_apply` public then. What else?

Reply:
It doesn't simplify much in terms of lines. I was mainly trying to avoid unfolding ex. But I think it is more like a style issue.